### ISSUE TITLE
Add `charset=utf-8` to text mime types

### ISSUE THREAD

**Issue by casey** [abonander/mime_guess#77]
I'm guessing mime types from extensions, but I know that whenever it's a text-based format, it's utf-8. Would it be possible to support guessing `charset=utf-8` where permissible? I.e. returning `text/html; charset=utf-8` when the path ends in `.html`?

**Comment by paolobarbolini** [comment]
I've been thinking about this same idea for a crate that depends on `mime_guess`, and I think it's incorrect to assume `charset=utf-8` just because the file is a text file and strings in Rust are utf-8, because you may be generating the mime type for a file generated by a _different_ application that doesn't have that.

**Comment by abonander** [comment]
Yeah, I don't think this is a good idea. HTML _can_ have a bunch of different possible encodings: https://en.wikipedia.org/wiki/Character_encodings_in_HTML#Permitted_encodings

It's not right for `mime_guess` to assume UTF-8.
